airflow/contrib/executors/kubernetes_executor.py

```diff
@@ -769,3 +793,7 @@ def _change_state(self, key, state, pod_id):
     def end(self):
         self.log.info('Shutting down Kubernetes executor')
         self.task_queue.join()
```

Comment:
I think here we had a problem here before because task_queue.join() was used but nowhere task_queue.task_done() was  executed. This could have resulted in hangs when shutting down Kubernetes Executor.